lib/Doctrine/Migrations/Configuration/ConfigurationLoader.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Migrations\Configuration;
+
+use Doctrine\Migrations\Configuration\Exception\UnknownLoader;
+use Doctrine\Migrations\Configuration\Loader\ArrayLoader;
+use Doctrine\Migrations\Configuration\Loader\JsonFileLoader;
+use Doctrine\Migrations\Configuration\Loader\Loader;
+use Doctrine\Migrations\Configuration\Loader\PHPFileLoader;
+use Doctrine\Migrations\Configuration\Loader\XmlFileLoader;
+use Doctrine\Migrations\Configuration\Loader\YamlFileLoader;
+use function count;
+
+class ConfigurationLoader
+{
+    /** @var Loader[] */
+    private $loaders = [];
+
+    public function addLoader(string $type, Loader $loader) : void
+    {
+        $this->loaders[$type] = $loader;
+    }
+
+    private function setDefaultLoaders() : void
+    {
+        $this->loaders = [
+            'array' => new ArrayLoader(),
```

Comment:
Is the order relevant here?

If not, what do you think about keeping this `array` sorted by key?